script.js

```diff
@@ -1,4 +1,216 @@
+      (inputSign == "stone" && resultOfComputerSign == paper) ||
+      (inputSign == "paper" && resultOfComputerSign == scissors)
+    ) {
+      //Add to User's win
+      userWin += 1;
+      //calling special message function
+      var specialMessageHere = specialMessage();
+      //calling gamemode message function
+      var gameModeMessageHere = gameModeMessage();
+      var myOutputValue =
+        "Computer selected " +
+        resultOfComputerSign +
+        " you win! <br> Try and Win again" +
+        username +
+        "! <br> Computer has so far won " +
+        computerWin +
+        " number of times. <br> You have so far won " +
+        userWin +
+        " number of times.. " +
+        specialMessageHere +
+        gameModeMessageHere;
+      return myOutputValue;
+    }
+    //Accounting for Lose scenario for Reversed version
+    if (
+      (inputSign == "scissors" && resultOfComputerSign == paper) ||
+      (inputSign == "stone" && resultOfComputerSign == scissors) ||
+      (inputSign == "paper" && resultOfComputerSign == stone)
+    ) {
+      //Add to Computer's win
+      computerWin += 1;
+      //calling special message function
+      var specialMessageHere = specialMessage();
+      //calling gamemode message function
+      var gameModeMessageHere = gameModeMessage();
+      var myOutputValue =
+        "Computer selected " +
+        resultOfComputerSign +
+        " you lose! <br> Try again" +
+        username +
+        "! <br> Computer has so far won " +
+        computerWin +
+        " number of times. <br> You have so far won " +
+        userWin +
+        " number of times. " +
+        specialMessageHere +
+        gameModeMessageHere;
+      return myOutputValue;
+    }
+    //Username assigning
+    username = inputSign;
+    var myOutputValue =
+      "Hi " +
+      username +
+      " Now please input either scissors, stone, or paper to start playing. Or 'reverse' or 'normal' to change modes!";
+    return myOutputValue;
+  }
+};
+
+//RandomNumber generator to assign Computer's SPS move
+var randomNumber = function () {
+  var randomDecimal = Math.random() * 3;
+  var randomInteger = Math.floor(randomDecimal);
+  var randomNumberResult = randomInteger + 1;
+  return randomNumberResult;
+};
+
+//Assigining Computer's SPS move
+var computerSign = function () {
+  var assignRandomNumberResult = randomNumber();
+  if (assignRandomNumberResult == 1) {
+    return scissors;
+  }
+  if (assignRandomNumberResult == 2) {
+    return stone;
+  }
+  return paper;
+};
+
+//formatting special message if user is losing/winning
+var specialMessage = function () {
+  if (computerWin == userWin) {
+    var myOutputValue = "You are currently tied with the computer! :|";
+    return myOutputValue;
+  }
+  if (computerWin < userWin) {
+    var myOutputValue = "You are ahead! :)";
+    return myOutputValue;
+  }
+  var myOutputValue = "You are behind! :(";
+  return myOutputValue;
+};
```

Comment:
Good use of helper functions here, in the gameModeMessage() function below and in the computerSign() function above. This is good practice in keeping your code readable, and will also help you keep track of the various gameModes and added functionalities!